Content.Server/Atmos/Piping/Unary/EntitySystems/GasVentPumpSystem.cs

```diff
@@ -104,9 +104,13 @@ private void OnGasVentPumpUpdated(EntityUid uid, GasVentPumpComponent vent, ref
                 if (pressureDelta <= 0)
                     return;
 
-                // how many moles to transfer to change external pressure by pressureDelta
-                // (ignoring temperature differences because I am lazy)
-                var transferMoles = pressureDelta * environment.Volume / (pipe.Air.Temperature * Atmospherics.R);
+                float transferMoles = 0;
+
+                // Don't transfer air if pressure is actively dropping
+                if (pressurizationLockout(vent, environment))
+                    // how many moles to transfer to change external pressure by pressureDelta
+                    // (ignoring temperature differences because I am lazy)
+                    transferMoles = pressureDelta * environment.Volume / (pipe.Air.Temperature * Atmospherics.R);
 
                 // Only run if the device is under lockout and not being overriden
                 if (vent.UnderPressureLockout & !vent.PressureLockoutOverride)
```

Comment:
Is `UnderPressureLockoutLeaking` still needed with this new implementation of UPLO?

Reply:
I'd say it is, currently my code is intended to run only if we're above `UnderPressureLockoutThreshold`. 
I think we should keep `UnderPressureLockoutLeaking` if we still want the eventual refilling of rooms. 

And now that I think about it, I seem to have forgotten to handle `PressureLockoutOverride`, though that should be just a single if-clause..

Reply:
I'm not sure. The reason this override was added was that the high lock-out meant that you couldn't re-pressurize from distro after the room was spaced. Using this new lock out, as long as the threshold was set slightly negative, then it should always repressurize itself, right? So the override would not be needed?

You make a good case for it to be left there for sabotage, though.